assets/home/tasks-section/index.js

```diff
@@ -1,12 +1,100 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Notice } from '@wordpress/components';
+import { useEffect, useState, useRef } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+
 /**
  * Internal dependencies
  */
+import { Col } from '../grid';
 import Section from '../section';
 import Progress from './progress';
 import Tasks from './tasks';
 import FirstCourse from './first-course';
 import Ready from './ready';
 
+/**
+ * Hook to update the state to ready when all tasks are completed.
+ *
+ * @param {boolean} isServerCompleted   Whether tasks were already completed in the server.
+ * @param {number}  totalTasks          Number of tasks.
+ * @param {number}  totalCompletedTasks Number of completed tasks.
+ */
+const useReadyState = (
+	isServerCompleted,
+	totalTasks,
+	totalCompletedTasks
+) => {
+	const [ ready, setReady ] = useState( isServerCompleted );
+	const [ readyError, setReadyError ] = useState( false );
+
+	useEffect( () => {
+		if ( ! ready && totalCompletedTasks === totalTasks ) {
+			apiFetch( {
+				path: '/sensei-internal/v1/home/tasks/complete',
+				method: 'POST',
+			} )
+				.then( () => {
+					setReady( true );
+				} )
+				.catch( ( err ) => {
+					setReadyError( err );
+				} );
+		}
+	}, [ totalTasks, totalCompletedTasks, ready ] );
+
+	return {
+		ready,
+		readyError,
+	};
+};
+
+/**
+ * Dismiss hook.
+ *
+ * @return {{dismissed:boolean, onDismiss:Function}} Object with dismissed and onDismiss callback.
+ */
+const useDismiss = () => {
+	const [ dismissed, setDismissed ] = useState( false );
+
+	const onDismiss = () => {
+		setDismissed( true );
+	};
+
+	return { dismissed, onDismiss };
+};
+
+/**
+ * Col preparation to height animation. Basically, it sets a max-height in the component based on
+ * its real height.
+ *
+ * @return {{colRef:Object, colStyle:Object}} Object with colRef and colStyle.
+ */
+const useColPreparationToHeightAnimation = () => {
+	const colRef = useRef();
+	const [ colStyle, setColStyle ] = useState( {} );
+
+	useEffect( () => {
+		if ( ! colRef.current ) {
+			return;
+		}
+
+		setColStyle( {
+			overflow: 'hidden',
+			maxHeight: colRef.current.offsetHeight,
```

Comment:
We need to set a `max-height` to the element, so we can animate it. Since this can be very different depending on the number of tasks or depending on the screen size, it gets the height dynamically.